### PR TITLE
new driver for Astromechanics LPM (light pollution meter) pro

### PR DESCRIPTION
**Pull request by chkettu**
An indi-lib driver for Astromechanics LPM (light pollution meter) PRO.

https://www.astromechanics.org/lpm.html

Find the documentation attached to this pull request. Feedback, improvements, etc. welcome.

[Astromechanics_LPM_Documentation.tar.gz]
(https://github.com/indilib/indi/files/3241811/Astromechanics_LPM_Documentation.tar.gz)

### CODE REVIEW

File: libindi/drivers/auxiliary/lpm.cpp [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+#include "connectionplugins/connectionserial.h"
+
+#include <cerrno>
+#include <memory>
+#include <cstring>
+#include <unistd.h>
+#include <sys/stat.h>
+
+// We declare an auto pointer to LPM.
+std::unique_ptr<LPM> lpm(new LPM());
```
Comment (knro) [inline libindi/drivers/auxiliary/lpm.cpp]: add static

File: libindi/drivers/auxiliary/lpm.cpp [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+            AverageReadingN[2].value = mpsas;
+            AverageReadingN[3].value = mpsas;
+        }
+    }
+    return true;
+}
+
+const char *LPM::getDefaultName()
+{
+    return (const char *)"LPM";
```
Comment (knro) [inline libindi/drivers/auxiliary/lpm.cpp]: remove (const char *), it's not required.

File: libindi/drivers/auxiliary/lpm.cpp [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+            count = 0;
+            return true;
+        } else if (!strcmp(name, SaveBP.name))
+        {
+            IUUpdateSwitch(&SaveBP, states, names, n);
+            snprintf(filename, 2048, "%s/%s", RecordFileT[0].text, RecordFileT[1].text);
+
+            if (SaveB[SAVE_READINGS].s == ISS_ON)
+            {
+                LOGF_INFO("Save readings to %s", filename);
```
Comment (knro) [inline libindi/drivers/auxiliary/lpm.cpp]: what happens when the user changes the file name? shouldn't the previous file be closed a new one be opened?

File: libindi/drivers.xml [hunk 4]
```diff
@@ -459,6 +459,10 @@
             <driver name="Weather Simulator">indi_simulator_weather</driver>
             <version>1.0</version>
         </device>
+        <device label="LPM">
+                <driver name="SQM">indi_astromech_lpm</driver>
```
Comment (knro) [inline libindi/drivers.xml]: name="LPM"

File: libindi/drivers.xml [hunk 5]
```diff
@@ -459,6 +459,10 @@
             <driver name="Weather Simulator">indi_simulator_weather</driver>
             <version>1.0</version>
         </device>
+        <device label="LPM">
```
Comment (knro) [inline libindi/drivers.xml]: label="Astromech LPM" is probably a more friendly label.

File: libindi/drivers/auxiliary/lpm.cpp (hunk 2 above)
Comment (TallFurryMan) [reply libindi/drivers/auxiliary/lpm.cpp]: Doesn't this even require a static storage? As is, it might not suit clang on OSX.

Comment (chkettu) [reply libindi/drivers/auxiliary/lpm.cpp]: removed cast - don't know, what "static storage" means in that context.

File: libindi/drivers/auxiliary/lpm.cpp [hunk 6]
```diff
@@ -32,6 +32,7 @@
 #include <cstring>
 #include <unistd.h>
 #include <sys/stat.h>
+#include <math.h>
 
 // We declare an auto pointer to LPM.
 std::unique_ptr<LPM> lpm(new LPM());
```
Comment (knro) [inline libindi/drivers/auxiliary/lpm.cpp]: static std::unique_ptr<LPM> lpm(new LPM());

**Comment by chkettu** [comment]
Documentation updated (added NELM-Measurement)

[Astromechanics_LPM_Documentation_1.tar.gz](https://github.com/indilib/indi/files/3244073/Astromechanics_LPM_Documentation_1.tar.gz)

File: libindi/drivers/auxiliary/lpm.cpp (hunk 2 above)
Comment (TallFurryMan) [reply libindi/drivers/auxiliary/lpm.cpp]: That's OK actually. I misread the prototype const for a non-const. Sorry.

**Comment by chkettu** [comment]
corrections of your requested changes done - please review again.

File: libindi/drivers/auxiliary/lpm.cpp [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+        }
+        //NELM (see http://unihedron.com/projects/darksky/NELM2BCalc.html)
+        AverageReadingN[4].value = 7.93-5*log(pow(10,(4.316-(mpsas/5)))+1)/log(10);
+    }
+    return true;
+}
+
+const char *LPM::getDefaultName()
+{
+    return "Astromechanics LPM";
```
Comment (knro) [inline libindi/drivers/auxiliary/lpm.cpp]: The same name should be used in the XML name attribute of the driver element.. so I suggest you update the XML file to:

```
<device label="Astromechanics LPM">
                <driver name="Astromechanics LPM">indi_astromech_lpm</driver>
                <version>0.1</version>
</device>
```

**Comment by knro** [comment]
Thanks. Documentation added: https://indilib.org/devices/auxiliary/astromechanics-light-pollution-meter-pro.html

**Pull request merged**